examples/camera-controller/webrtc_manager/WebRTCManager.cpp

```diff
@@ -113,29 +120,32 @@ CHIP_ERROR WebRTCManager::Connnect(Controller::DeviceCommissioner & commissioner
         ChipLogProgress(Camera, "[Gathering State: %d]", static_cast<int>(state));
     });
 
-    // Create a data channel for this offerer
-    mDataChannel = mPeerConnection->createDataChannel("test");
-
-    if (mDataChannel)
-    {
-        mDataChannel->onOpen(
-            [&]() { ChipLogProgress(Camera, "[DataChannel open: %s]", mDataChannel ? mDataChannel->label().c_str() : "unknown"); });
-
-        mDataChannel->onClosed([&]() {
-            ChipLogProgress(Camera, "[DataChannel closed: %s]", mDataChannel ? mDataChannel->label().c_str() : "unknown");
-        });
-
-        mDataChannel->onMessage([](auto data) {
-            if (std::holds_alternative<std::string>(data))
-            {
-                ChipLogProgress(Camera, "[Received: %s]", std::get<std::string>(data).c_str());
-            }
-        });
-    }
+    mMedia = rtc::Description::Video("video", rtc::Description::Direction::RecvOnly);
+    mMedia.addH264Codec(96);
+    mMedia.setBitrate(3000);
+    mTrack = mPeerConnection->addTrack(mMedia);
+
+    mDepacketizer = std::make_shared<rtc::H264RtpDepacketizer>();
+    mTrack->setMediaHandler(mDepacketizer);
+    mTrack->onFrame([this](rtc::binary message, rtc::FrameInfo frameInfo) {
+        ChipLogProgress(Camera, "Sending H264 Frame");
+        sendto(sock, reinterpret_cast<const char *>(message.data()), int(message.size()), 0,
+               reinterpret_cast<const struct sockaddr *>(&socket_address), sizeof(socket_address));
+    });
+    mPeerConnection->setLocalDescription();
 
     return CHIP_NO_ERROR;
 }
 
+void WebRTCManager::initializeSocket()
+{
+    sock                           = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
Add error handling for the socket call and maybe return that from initializeSocket()

Reply:
Thank you for your review, We fixed your comments